pkg/profiling/profiling_suite_test.go

```diff
@@ -0,0 +1,48 @@
+package profiling_test
+
+import (
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var testenv *envtest.Environment
+var cfg *rest.Config
+var clientset *kubernetes.Clientset
+
+func TestProfiling(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Profiling Suite")
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	testenv = &envtest.Environment{}
+
+	var err error
+	cfg, err = testenv.Start()
+	Expect(err).NotTo(HaveOccurred())
+
+})
+
+func getFreePort() (string, error) {
+	addr, err := net.ResolveTCPAddr("tcp", ":0")
```

Comment:
I'd like to suggest changing this line with:

```golang
addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
```

For the purpose of not getting a pop up in Mac asking if you want to permit bla bla bla firewall non-nonse.